src/components/media-loader.js

```diff
@@ -257,13 +259,38 @@ AFRAME.registerComponent("media-loader", {
     }
   },
 
+  refresh() {
+    if (this.networkedEl && !NAF.utils.isMine(this.networkedEl) && !NAF.utils.takeOwnership(this.networkedEl)) return;
+
+    // When we refresh, we bump the version to the current timestamp.
+    //
+    // The only use-case for refresh right now is re-fetching screenshots.
+    this.el.setAttribute("media-loader", { version: Math.floor(Date.now() / 1000) });
+  },
+
   async update(oldData) {
+    const { src, version } = this.data;
+
+    if (!src) return;
+    const srcChanged = oldData.src !== src;
+    const versionChanged = !!(oldData.version && oldData.version !== version);
+
+    if (versionChanged) {
+      this.el.emit("media_refreshing");
+
+      // Don't animate if its a refresh.
+      this.data.animate = false;
+
+      // Play the sound effect on a refresh only if we are the owner
+      this.data.playSoundEffect = NAF.utils.isMine(this.networkedEl);
+    }
+
     try {
-      const { src, contentSubtype } = this.data;
+      const { src, version, contentSubtype } = this.data;
```

Comment:
src and version are already declared in this scope.

Reply:
ah whoops, had extracted this to a function and reverted but left this bit.